src/prepare.js

```diff
@@ -179,6 +185,111 @@ function getConfigOverride(confStr) {
 	return config;
 }
 
+/**
+ * @param {ReturnType<getConfigOverride>} conf
+ * @param {ReturnType<typeof getBasicBuildOptions>["toolchain"]} toolchain
+ */
+async function extendConfig(conf, toolchain) {
+	if (!conf) return;
+	if (toolchain === "respec" && !conf.specStatus) return;
+	if (toolchain === "bikeshed" && !conf.status) return;
+
+	/** Get present date in YYYY-MM-DD format */
+	const getShortIsoDate = () => new Date().toISOString().slice(0, 10);
+
+	let publishDate = getShortIsoDate();
+	if (toolchain === "respec") {
+		conf.publishDate = publishDate = conf.publishDate || publishDate;
+	} else if (toolchain === "bikeshed") {
+		conf.date = publishDate = conf.date || publishDate;
+	}
+
+	const shortName = conf.shortName || conf.shortname;
+	if (shortName) {
+		const prev = await getPreviousVersionInfo(shortName, publishDate);
+		if (toolchain === "respec") {
+			conf.previousMaturity = prev.previousMaturity;
+			conf.previousPublishDate = prev.previousPublishDate;
+		} else if (toolchain === "bikeshed") {
+			conf["previous version"] = prev.previousURI;
+		}
+	}
+}
+
+/**
+ * @param {string} shortName
+ * @param {string} publishDate
+ */
+async function getPreviousVersionInfo(shortName, publishDate) {
+	console.group(`[INFO] Finding previous version details...`);
+	const url = "https://www.w3.org/TR/" + shortName + "/";
+
+	const browser = await puppeteer.launch({
+		executablePath: PUPPETEER_ENV.PUPPETEER_EXECUTABLE_PATH,
+	});
+
+	try {
+		const page = await browser.newPage();
+		console.log("[INFO] Navigating to", url);
+		const res = await page.goto(url);
+		console.log("[INFO] Navigation complete with statusCode:", res.status());
+		if (!res.ok) {
+			throw new Error(`Failed to fetch ${url}`);
+		}
+
+		const thisURI = await page.$$eval("body div.head dl dt", elems => {
+			const thisVersion = elems.find(el =>
+				/this (?:published )?version/i.test(el.textContent.trim()),
+			);
+			if (thisVersion) {
+				const dd = thisVersion.nextElementSibling;
+				if (dd && dd.localName === "dd") {
+					return dd.querySelector("a").href;
+				}
+			}
+			return null;
+		});
+		console.log("[INFO] thisURI:", thisURI);
+
+		const previousURI = await page.$$eval("body div.head dl dt", elems => {
+			const previousVersion = elems.find(el =>
+				/previous (?:published )?version/i.test(el.textContent.trim()),
+			);
+			if (previousVersion) {
+				const dd = previousVersion.nextElementSibling;
+				if (dd && dd.localName === "dd") {
+					return dd.querySelector("a").href;
+				}
+			}
+			return null;
+		});
+		console.log("[INFO] prevURI:", previousURI);
+
+		if (!thisURI) {
+			throw new Error(
+				"Couldn't find a 'This version' uri in the previous version.",
+			);
+		}
+
+		const thisDate = thisURI.match(/[1-2][0-9]{7}/)[0];
+		const prevDate = publishDate.replace(/\-/g, "");
```

Comment:
kind of a nit, but calling this `prevDate` is pretty confusing if I understand the code correctly

Reply:
`currentTrDate`?

Reply:
if I understand this right, is the date of the targetted TR publication - so maybe, `targetPublishDate`